tests/test_application.py

```diff
@@ -239,3 +239,44 @@ def tester(url, data=None, **kwargs):
 
     # Will not test scenario of app leaving family. Per specs, it won't happen.
 
+class TestClientApplicationforAuthorityMigration(unittest.TestCase):
+
+    @classmethod
+    def setUp(self):
+        self.authority_url = "https://sts.windows.net/common"
+        self.scopes = ["s1", "s2"]
+        self.uid = "uid"
+        self.utid = "utid"
+        self.account = {"home_account_id": "{}.{}".format(self.uid, self.utid)}
+        self.access_token = "access_token"
+        self.refresh_token = "refresh_token"
+        self.cache = msal.SerializableTokenCache()
+        self.cache.add({
+            "client_id": "my_client_id",
+            "scope": self.scopes,
+            "token_endpoint": "{}/oauth2/v2.0/token".format(self.authority_url),
+            "response": TokenCacheTestCase.build_response(
+                uid=self.uid, utid=self.utid, access_token=self.access_token, refresh_token=self.refresh_token),
+
+        })  # The add(...) helper populates correct home_account_id for future searching
+
+    def test_get_accounts(self):
+        app = PublicClientApplication(
+            "my_client_id", authority="https://login.microsoftonline.com/common",
+            token_cache=self.cache)
+
+        accounts = app.get_accounts()
+        self.assertEqual({'home_account_id': 'uid.utid',
+                          'environment': 'sts.windows.net',
+                          'realm': 'common',
+                          'local_account_id': None,
+                          'username': None,
+                          'authority_type': 'MSSTS'}, accounts[0])
+
+    def test_acquire_token_silent(self):
+        app = PublicClientApplication(
+            "my_client_id", authority="https://login.microsoftonline.com/common",
+            token_cache=self.cache)
+        a = app.acquire_token_silent(["s1", "s2"], self.account)
```

Comment:
Ask me a funny story about this variable naming. :)

Reply:
Hey Ray, is there a reason why its different than the one we add in token_cache? The acquire token silent will fail if its a different client_id causing the test case to fail ?

Reply:
Nope, client_id should remain the same here.

In fact, in some earlier version there were some hardcoded client_id here and there, thus vulnerable to fragile test behaviors due to accidentally changing some of the client_id. So in the current version, we define one single copy of truth as `self.client_id` and reuse them everywhere. That's the whole point of defining some test settings as class-wide members. (On the other hand, the current version also demote some previous `self.xyz` settings to local variables if they are only used inside `setUp()`.)